pkg/porter/version_test.go

```diff
@@ -51,3 +54,30 @@ func TestPrintJsonVersion(t *testing.T) {
 		t.Fatalf("invalid output:\nWANT:\t%q\nGOT:\t%q\n", wantOutput, gotOutput)
 	}
 }
+
+func TestPrintSystemDebugInfo(t *testing.T) {
+	pkg.Commit = "abc123"
+	pkg.Version = "v1.2.3"
+
+
+	p := NewTestPorter(t)
+
+	opts := version.Options{}
+	p.TestConfig.SetupPorterHome()
+	err := opts.Validate()
+	require.Nil(t, err)
+	err = p.PrintDebugInfo(opts)
```

Comment:
After you move this method and the check for `opts.System` into `p.PrintVersion`, I recommend updating this unit test, to exercise your if statement as well. So it should create `VersionOptions {System: true}` and pass that to `p.PrintVersion(opts)`.